pyomo/contrib/interior_point/interior_point.py

```diff
@@ -0,0 +1,669 @@
+            objective = interface.evaluate_objective()
+            self.logger.info('{_iter:<6}'
+                             '{objective:<11.2e}'
+                             '{primal_inf:<11.2e}'
+                             '{dual_inf:<11.2e}'
+                             '{compl_inf:<11.2e}'
+                             '{barrier:<11.2e}'
+                             '{alpha_p:<11.2e}'
+                             '{alpha_d:<11.2e}'
+                             '{reg:<11.2e}'
+                             '{time:<7.3f}'.format(_iter=_iter,
+                                                   objective=objective,
+                                                   primal_inf=primal_inf,
+                                                   dual_inf=dual_inf,
+                                                   compl_inf=complimentarity_inf,
+                                                   barrier=self._barrier_parameter,
+                                                   alpha_p=alpha_primal_max,
+                                                   alpha_d=alpha_dual_max,
+                                                   reg=reg_coef,
+                                                   time=time.time() - t0))
+
+            if max(primal_inf, dual_inf, complimentarity_inf) <= tol:
+                break
+            timer.start('convergence check')
+            primal_inf, dual_inf, complimentarity_inf = \
+                    self.check_convergence(barrier=self._barrier_parameter, timer=timer)
+            timer.stop('convergence check')
+            if max(primal_inf, dual_inf, complimentarity_inf) \
+                    <= 0.1 * self._barrier_parameter:
+                # This comparison is made with barrier problem infeasibility.
+                # Sometimes have trouble getting dual infeasibility low enough
+                self.update_barrier_parameter()
+
+            interface.set_barrier_parameter(self._barrier_parameter)
+            timer.start('eval')
+            timer.start('eval kkt')
+            kkt = interface.evaluate_primal_dual_kkt_matrix(timer=timer)
+            timer.stop('eval kkt')
+            timer.start('eval rhs')
+            rhs = interface.evaluate_primal_dual_kkt_rhs(timer=timer)
+            timer.stop('eval rhs')
+            timer.stop('eval')
+
+            # Factorize linear system
+            timer.start('factorize')
+            reg_coef = self.factorize(kkt=kkt, timer=timer)
+            timer.stop('factorize')
+
+            timer.start('back solve')
+            with self.linear_solve_context:
+                self.logger.info('Iter: %s' % self._iter)
+                delta = linear_solver.do_back_solve(rhs)
+            timer.stop('back solve')
+
+            interface.set_primal_dual_kkt_solution(delta)
+            timer.start('frac boundary')
+            alpha_primal_max, alpha_dual_max = \
+                    self.fraction_to_the_boundary()
```

Comment:
No line search?

Reply:
Not yet. Starting simple. We'll get there.